src/rpc/rawtransaction.cpp

```diff
@@ -804,14 +805,30 @@ UniValue rawblindrawtransaction(const UniValue& params, bool fHelp)
 
     // How many are we trying to blind?
     int numPubKeys = 0;
+    CPubKey nonZeroKey;
     for (auto&& key : output_pubkeys) {
         if (key.IsValid()) {
             numPubKeys++;
+            nonZeroKey = key;
         }
     }
 
-    if (numPubKeys == 0 || BlindTransaction(input_blinds, input_asset_blinds, input_assets, input_amounts, output_value_blinds, output_asset_blinds, output_pubkeys, std::vector<CKey>(), std::vector<CKey>(), tx) != numPubKeys) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, string("Unable to blind transaction: add an additional output with a blinding pubkey"));
+    CMutableTransaction txBackup(tx);
+    if (numPubKeys == 0) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, string("Transaction has no outputs to blind"));
+    }
+    if (BlindTransaction(input_blinds, input_asset_blinds, input_assets, input_amounts, output_value_blinds, output_asset_blinds, output_pubkeys, std::vector<CKey>(), std::vector<CKey>(), tx) != numPubKeys) {
```

Comment:
This should be similar to what we're doing in blindraw: check for the conditions to add an OP_RETURN directly.